### PR TITLE
[Fix][Text Generation Pipeline] Fix the erroneous sampling logic

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Fix Description

Before: regardless of whether `sampling=True or False` we would do `top_k `and `top_p` sampling.
Now: if `sampling=False`, we directly "jump" to the argmax function and avoid any sampling logic.

@horheynm Could you please validate the rest of the logic in `def generate(self, logits: numpy.ndarray)`? In the most complex scenario, we can apply both `top_k`, `top_p`, and `sampling_temperature` sequentially to our logits. Let's make sure that the order in which the sampling functions are applied matches the one defined in HF (I assume this is the original implementation that we want to mimic).

### CODE REVIEW

File: src/deepsparse/transformers/utils/token_generator.py [hunk 1]
```diff
@@ -77,16 +77,17 @@ def generate(self, logits: numpy.ndarray) -> numpy.ndarray:
         :param logits: the logits from the model with shape (vocab_size,)
         :return: the sampled token
         """
-        if self.top_k:
-            logits = self.apply_top_k(logits)
-        if self.top_p:
-            logits = self.apply_top_p(logits)
-
```
Comment (dsikka) [inline src/deepsparse/transformers/utils/token_generator.py:84]: Seems to be the same case for v2?

Comment (dbogunowicz) [reply src/deepsparse/transformers/utils/token_generator.py:84]: Yes. This fix makes the e2e tests for v2 green :)

Comment (dsikka) [reply src/deepsparse/transformers/utils/token_generator.py:84]: Nice!

**Comment by rahul-tuli** [comment]
Could you add output before and after?

**Review by bfineran — APPROVED** [review]
